### PR TITLE
feat: add finite difference solver

### PR DESCRIPTION
**Pull request by vishwa2710**

### CODE REVIEW

File: include/OpenSpaceToolkit/Astrodynamics/Solvers/FiniteDifferenceSolver.hpp [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+    {
+        Forward,
+        Backward,
+        Central
+    };
+
+    /// @brief                  Constructor
+    ///
+    /// @code
+    ///                         const Real stepPercentage = 1e-3;
```
Comment (kyle-cochran) [inline include/OpenSpaceToolkit/Astrodynamics/Solvers/FiniteDifferenceSolver.hpp:49]: Is this code example correct for this one? Looks like it takes a type instead

File: include/OpenSpaceToolkit/Astrodynamics/Solvers/FiniteDifferenceSolver.hpp [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+    /// @param                  [in] anInstant An instant.
+    /// @param                  [in] getState Callable to generate a State at the requested Instant.
+    /// @param                  [in] aStepPercentage The step percentage to use for the perturbation. Defaults to 1e-3.
+    ///
+    /// @return                 The State Transition Matrix
+
+    MatrixXd computeStateTransitionMatrix(
+        const State& aState,
+        const Instant& anInstant,
+        std::function<State(const State&, const Instant&)> getState,
```
Comment (kyle-cochran) [inline include/OpenSpaceToolkit/Astrodynamics/Solvers/FiniteDifferenceSolver.hpp:95]:
```suggestion
        std::function<State(const State&, const Instant&)> propagateState,
```

nit: a bit pedantic, but something like `propagateState` (or `getStateAtInstant` or `propagateStateToInstant`) is a little clearer IMO and describes what the input lambda should do.

**Review by kyle-cochran — APPROVED** [review]
Couple nits. lgtm

File: include/OpenSpaceToolkit/Astrodynamics/Solvers/FiniteDifferenceSolver.hpp (hunk 2 above)
Comment (vishwa2710) [reply include/OpenSpaceToolkit/Astrodynamics/Solvers/FiniteDifferenceSolver.hpp]: I've updated this to generateState, agreed that getState is a bit generic and used as getter.

**Pull request merged**